src/video/video_source_local.rs

```diff
@@ -159,48 +182,87 @@ impl VideoSource for VideoSourceLocal {
-                for v4l_size in v4l_framesizes.size.to_discrete() {
-                    match &device.enum_frameintervals(
-                        v4l_framesizes.fourcc,
-                        v4l_size.width,
-                        v4l_size.height,
-                    ) {
-                        Ok(enum_frameintervals) => {
-                            let intervals = convert_v4l_intervals(enum_frameintervals);
-                            sizes.push(Size {
-                                width: v4l_size.width,
-                                height: v4l_size.height,
-                                intervals,
-                            })
-                        }
-                        Err(error) => {
-                            warn!("Failed to fetch frameintervals for camera: {}, for encode: {:?}, for size: {:?}, error: {:#?}",
-                                &self.device_path, v4l_format.fourcc, v4l_size, error);
+                match v4l_framesizes.size {
+                    v4l::framesize::FrameSizeEnum::Discrete(v4l_size) => {
+                        match &device.enum_frameintervals(
+                            v4l_framesizes.fourcc,
+                            v4l_size.width,
+                            v4l_size.height,
+                        ) {
+                            Ok(enum_frameintervals) => {
+                                let intervals = convert_v4l_intervals(enum_frameintervals);
+                                sizes.push(Size {
+                                    width: v4l_size.width,
+                                    height: v4l_size.height,
+                                    intervals: intervals.into(),
+                                })
+                            }
+                            Err(error) => {
+                                warn!("Failed to fetch frameintervals for camera: {}, for encode: {:?}, for size: {:?}, error: {:#?}",
+                                    &self.device_path, v4l_format.fourcc, v4l_size, error);
+                            }
                         }
                     }
+                    v4l::framesize::FrameSizeEnum::Stepwise(v4l_size) => {
+                        let std_sizes: Vec<(u32, u32)> = vec![
+                            (v4l_size.max_width, v4l_size.max_height),
+                            (7200, 3060),
+                            (7680, 4320),
+                            (3840, 2160),
+                            (2560, 1440),
+                            (1440, 1080),
+                            (1920, 1080),
+                            (1280, 1080),
+                            (1600, 1200),
+                            (1280, 720),
+                            (1024, 768),
+                            (960, 720),
+                            (800, 600),
+                            (640, 480),
+                            (640, 360),
+                            (320, 240),
+                            (256, 144),
```

Comment:
I've chosen to only add common and widely used resolutions as opposed to deciding resolutions algorithmically because I found it to be near impossible to balance the number of sizes with how useful they would actually be, every resolution looks random, like `1326x1768` for 4:3, and if we would be adding the standard ones, why should we even bother trying that...